### PR TITLE
Altered config parsing

### PR DESCRIPTION
**Pull request by Hiyorimi**

### CODE REVIEW

**Comment by Hiyorimi** [comment]
@theskumar I've added Boolean type support and improved quotation handling. 

Can you merge my pull request please!

File: dotenv.py [hunk 1]
```diff
@@ -86,7 +86,12 @@ def parse_dotenv(dotenv_path):
             if not line or line.startswith('#') or '=' not in line:
                 continue
             k, v = line.split('=', 1)
-            v = v.strip("'").strip('"')
+            v = str(v).replace("'", "").replace('"', '').replace(" ", "")
```
Comment (theskumar) [inline dotenv.py]: @Hiyorimi The logic earlier was to strip the leading quotes and don't modify the inner string of the value. so `foo="bar"` or `foo="a bar"` will be `bar` and `a bar`.  

As for the quotes inside the value or key, we shouldn't modify/remove them, as max an warning or error can be displayed.

File: dotenv.py [hunk 2]
```diff
@@ -86,7 +86,12 @@ def parse_dotenv(dotenv_path):
...
+            if v.lower() == 'true':
+                v = True
+            elif v.lower() == 'false':
+                v = False
```
Comment (theskumar) [inline dotenv.py]: We can't support type conversion as the goal of the this library to only populate the environment with variables in the same way it might have been added by any other method. So any other tool reading the `os.environ` always works with the expectation that it's going to be string. 

TLDR; Type conversions must be explicit. And it should be responsibility of the tool reading the values.

**Comment by hugochinchilla** [comment]
@Hiyorimi take a look at [python-decouple](https://pypi.python.org/pypi/python-decouple), it does what you need, and it pairs well with python-dotenv

**Comment by Hiyorimi** [comment]
@hugochinchilla thx, I will take a look.
